src/renderer/api/kube-object.ts

```diff
@@ -65,6 +66,42 @@ export class KubeStatus {
 
 export type IKubeMetaField = keyof IKubeObjectMetadata;
 
+/**
+ * This represents a precondition that must be satisfied before an edit is allowed
+ */
+export interface EditPrecondtion {
+  /**
+   * Checks to see if the edited version matches the original. Should return
+   * `true` if it passes the check.
+   * @param original The original version of the kube object
+   * @param edited The edited version of the kube object
+   */
+  check(original: KubeObject, edited: KubeObject): boolean;
+
+  /**
+   * This is error message that will be shown if the check fails.
+   */
+  message: string;
+}
+
+function requiredKeyUnchanged(key: keyof KubeObject): EditPrecondtion {
+  return {
+    check: (original: KubeObject, edited: KubeObject): boolean => (
+      deepEqual(edited[key], original[key], { strict: true })
+    ),
+    message: `${key} has changed`,
+  };
+}
+
+function requiredMetadataKeyUnchanged(key: keyof KubeObject["metadata"]): EditPrecondtion {
+  return {
+    check: (original: KubeObject, edited: KubeObject): boolean => (
+      deepEqual(edited.metadata[key], original.metadata[key], { strict: true })
+    ),
+    message: `metadata.${key} has changed`,
+  };
+}
```

Comment:
This functions looks very similar.  Can we keep just one?

Also I would propose better following definition `checkForErrorsBeforeUpdate(current: KubeObject, changed: Partial<KubeObject>)`. Wdyt?

And why do we need this at all? If I try to modify these fields wouldn't k8s-api respond with an error if it's not allowed?

Reply:
No the k8s-api doesn't do that unfortunately. This is actually what `kubectl` does when you try and edit something.

Do you mean instead of having it modular like this? I sort of like how this can be expanded rather easily in the future if we need to.

As for having only one, sure that isn't too hard.